testsuite/features/step_definitions/command_steps.rb

```diff
@@ -283,40 +283,38 @@
   "\n-----\n#{output[:stderr]}\n-----\n"
 end
 
-When(/^I check the tomcat logs for errors$/) do
+Then(/^the tomcat logs should not contain errors$/) do
   output = $server.run('cat /var/log/tomcat/*')
   msgs = %w[ERROR NullPointer]
   msgs.each do |msg|
     raise "-#{msg}-  msg found on tomcat logs" if output.include? msg
   end
 end
 
-Then(/^I restart the spacewalk service$/) do
+When(/^I restart the spacewalk service$/) do
   $server.run('spacewalk-service restart')
 end
 
-Then(/^I shutdown the spacewalk service$/) do
+When(/^I shutdown the spacewalk service$/) do
   $server.run('spacewalk-service stop')
 end
 
-Then(/^I execute spacewalk-debug on the server$/) do
+When(/^I execute spacewalk-debug on the server$/) do
   $server.run('spacewalk-debug')
-  cmd = "echo | scp -o StrictHostKeyChecking=no root@#{$server.ip}:/tmp/spacewalk-debug.tar.bz2 . 2>&1"
-  command_output = `#{cmd}`
-  unless $CHILD_STATUS.success?
-    raise "Execute command failed: #{$ERROR_INFO}: #{command_output}"
-  end
+  code = file_extract($server, "/tmp/spacewalk-debug.tar.bz2", "spacewalk-debug.tar.bz2")
```

Comment:
Are you sure this is working? I tried it and failed. This was the reason why I used `scp`

Reply:
It had been tested and worked. I got the file `spacewalk-debug.tar.bz2` on the controller.

I will now try to uncompress it to check its integrity.

Reply:
I think I will revert that one, just for safety. It could be it has problems with big files (?) .